add facing direction and sprite flipping to LivingEntity objects

LeoLamCY opened the pull request:
-add facing direction to LivingEntity objects
-add flipping of LivingEntity sprites according to the direction they are moving

jmrapp1 commented:
There are a few things that need to be changed with this implementation. I don't think it's necessary for the `AtlasRegionDrawable`, `Drawable`, `SpriteDrawable`, and `TexturedDrawable` classes to have a new `render` method taking a `flipped` parameter. Because the actual logic of the `flipped` parameter can be an instance variable in all `LivingEntity`'s, the only thing that needs to be changed is the `LivingEntity`'s render method. No new parameters are needed, all you need to do is move the logic from the `TexturedDrawable` new `render` method that you have added, and move it to the render method within the `LivingEntity` class. This way all implementations of `LivingEntity` will take this into account and there will be no need for any change in logic outside of the class. 

Small thing, would you be able to switch the `Direction` class to an `Enum` instead?

Also, when it comes to changing the `flipped` value of a `LivingEntity` when it moves, because this will need to be apart of every `LivingEntity` implementation and we know that every `LivingEntity` will need to be flipped in the direction that they are being moved, all that we need to do is modify the methods that move the entity to flip them a well. So instead of adding the logic in the `GameHud` to flip the entity, it would be better to move that logic to the `addVelocity` method within the `LivingEntity` class itself. This way every single `LivingEntity` will automatically flip and there will not need to be any separate logic for each entity.

LeoLamCY commented:
I have made some changes according to your suggestions.
I have moved the direction changing code to `addVelocity()` in `LivingEntity` but it's not perfect. The Player does not face the other way immediately after the movement control pad is pushed to the opposite direction, which I think it should. The Player only flips around once its velocity in the original direction reaches 0 and starts increasing in the opposite direction.
I haven't had time to really look through your code to understand why it's doing that. There might be better ways to do it, maybe you could provide some insights.

As for why I didn't make `Direction` an `Enum`:
[https://www.youtube.com/watch?v=Hzs6OBcvNQE](https://www.youtube.com/watch?v=Hzs6OBcvNQE)
I have now removed the class and turned it into constants in `LivingEntity` instead.

jmrapp1 commented:
It's very strange that it wouldn't flip automatically. I'll pull down the code and test it myself to see why. 

As for the `Enum`'s taking more memory, it's very strange that this is the way the compiler works. Google saying "not to use enums" takes away some code quality. If they're in charge of the compiler, I'm confused as to why they don't just break enums down into integers from within it. Very strange, but because this is how it works, we definitely don't want our memory running away because of a few enums.
